.NET/Microsoft.Recognizers.Text.DateTime/English/Parsers/EnglishHolidayParserConfiguration.cs

```diff
@@ -145,6 +161,238 @@ public override string SanitizeHolidayToken(string holiday)
 
         private static DateObject Veteransday(int year) => new DateObject(year, 11, 11);
 
-        private static DateObject EasterDay(int year) => DateObject.MinValue;
+        private static DateObject EasterDay(int year) => GetEasterYearList().Contains(year) ? GetEasterDateList().Where(d => d.Year == year).First() : DateObject.MinValue;
+
+        private static DateObject AshWednesday(int year) => EasterDay(year).AddDays(-46);
+
+        private static DateObject PalmSunday(int year) => EasterDay(year).AddDays(-7);
+
+        private static DateObject MaundyThursday(int year) => EasterDay(year).AddDays(-3);
+
+        private static DateObject GoodFriday(int year) => EasterDay(year).AddDays(-2);
+
+        private static DateObject EasterSaturday(int year) => EasterDay(year).AddDays(-1);
+
+        private static DateObject EasterMonday(int year) => EasterDay(year).AddDays(1);
+
+        private static DateObject AscensionDay(int year) => EasterDay(year).AddDays(39);
+
+        private static DateObject WhiteSunday(int year) => EasterDay(year).AddDays(49);
+
+        private static DateObject WhiteMonday(int year) => EasterDay(year).AddDays(50);
+
+        private static DateObject TrinitySunday(int year) => EasterDay(year).AddDays(56);
+
+        private static DateObject CorpusChristi(int year) => EasterDay(year).AddDays(60);
+
+        private static IList<int> GetEasterYearList() => GetEasterDateList().Select(d => d.Year).ToList();
+
+        // easter sunday: 1900 ~ 2099, following the Paschal Full Moon (PFM) date of lunar year calendar.
+        private static IList<DateObject> GetEasterDateList()

```

Comment:
Doesn't German has the lunar year calculation stuff? If so, was there an issue re-using it?
If not, can't this be in the YAML file somehow?

Reply:
Could you point me to the Lunar implementation in German? Thanks

Reply:
Thanks for pointing the German implementation. Tried to generalize the 'easter' family into base class, but got conflicts from other languages e.g. Dutch, so stepped back and just adopted the function from German into English for now.